Support other fields that don't end in input

jsteiner opened the pull request:
There are a handful of other fields that don't end in input. This adds
support for some of them.

This is still missing support for `radio_button` and `multiple_select`.
The former will require some additional code changes. The later may
require some additional considerations about the API. (SimpleForm
supports it as `collection`.

drapergeek commented:
Have you verified that the options pass down properly? I know some of them have interesting interfaces.

jsteiner commented:
All the ones I've added take `form, field, opts`. It looks like `radio_button`, `select`, and `multi_select` are different, but I didn't touch those.

I've added some tests to cover what I added.
